Pod/Classes/Optional Data Sources/VOKPagingFetchedResultsDataSource.m

```diff
@@ -158,32 +231,41 @@ - (void)scrollViewDidScroll:(UIScrollView *)scrollView
 
 - (void)triggerAction:(VOKPagingResultsAction)action
      forAccessoryView:(UIView<VOKPagingAccessory> *)accessory
-           withInsets:(UIEdgeInsets)insets
+    withContentOffset:(CGPoint)contentOffset
+    returningToOffset:(CGPoint)returnOffset
+      temporaryInsets:(UIEdgeInsets)tempInsets
 {
     self.isLoading = YES;
     [self.tableView setUserInteractionEnabled:NO];
-    
+
+    UIEdgeInsets originalInsets = self.tableView.contentInset;
     [accessory loadingWillBegin];
-    
-    [UIView animateWithDuration:.3 animations:^{
-        [self.tableView setContentInset:insets];
-    } completion:^(BOOL finished) {
-        VOKCompletionAction completionAction = ^void (void)
-        {
-            self.isLoading = NO;
-            [accessory loadingHasFinished];
-            
-            [self.tableView setUserInteractionEnabled:YES];
-            [self.tableView setContentInset:self.orginalInsets];
-        };
-        
-        action(self.tableView, completionAction);
-    }];
+
+    //self.table is a weak relationship, causing it to be dealloc'd before self while this block was in-flight.
+    typeof(self.tableView) __strong strongTableView = self.tableView;
+    [UIView animateWithDuration:.3
+                     animations:^{
+                         strongTableView.contentInset = tempInsets;
+                         [strongTableView setContentOffset:contentOffset];
+                     }
+                     completion:^(BOOL finished) {
+                         VOKCompletionAction completionAction = ^void (void)
+                         {
```

Comment:
```objc
VOKCompletionAction completionAction = ^{
```
and maybe have the block in-line where it's used instead of defining a local variable?